frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/ShareButton/utils.tsx

```diff
@@ -117,8 +163,6 @@ export const loadStateFromQueryParams = (
   const newFilteredCellTypes = params.get("cellTypes")?.split(delimiter) || [];
   if (newFilteredCellTypes.length > 0) paramsToRemove.push("cellTypes");
 
-  removeParams(paramsToRemove);
```

Comment:
this is the bug where `removeParams` is called before `"compare"` is pushed in `paramsToRemove`, which is why compare param is never removed from the URL. I've added a note below as a reminder